src/components/AppRoute.jsx

```diff
@@ -9,7 +9,7 @@ export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)
 const AppRoute = (
   <Route component={App}>
     <Redirect from='/' to='photos' />
-    <Route path='photos' component={Timeline} />
+    <Route path='photos(/:photoId)' component={Timeline} />
```

Comment:
Using a nested route here would allow to simplify the Timeline component (see my next comment):

```jsx
<Route path='photos' component={Timeline}>
  <Route path=':photoId' component={Viewer} />
</Route>
```